lib/exercises.rb

```diff
@@ -1,19 +1,44 @@
 
 # This method will return an array of arrays.
 # Each subarray will have strings which are anagrams of each other
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: o(n)
 
 def grouped_anagrams(strings)
-  raise NotImplementedError, "Method hasn't been implemented yet!"
+  # raise NotImplementedError, "Method hasn't been implemented yet!"
+  return [] if strings.empty?
+    hash = Hash.new([])
+    strings.each.with_index do |element, i| 
+      hash[element.downcase.chars.sort.join] += [i] 
+    end
+    # p hash
+    results = hash.map do |key, values| 
+      values.map do |i| 
+        strings[i]
+      end
+    end
```

Comment:
This nested loop will still only loop through `n` times because the size of the 2D array is only going to have `n` elements.